### PR TITLE
theme designer: fix tab states

### PR DESCRIPTION
**Pull request by emmayjiang**
Added a local state to sidebar that allows for the custom theme to be stored, and also so that switching tabs/switching themes updates correctly.

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/5e326ed2e7068e8eb75df4a54495c27206ec831b'>5e326ed2e7068e8eb75df4a54495c27206ec831b</a></sub>
<div id="bundleSizeComment9423"/>

File: packages/react-components/theme-designer/src/components/Sidebar/EditTab.tsx [hunk 1]
```diff
@@ -41,32 +41,25 @@ const useStyles = makeStyles({
...
   dispatchState: React.Dispatch<DispatchTheme>;
+  formState: CustomAttributes;
+  setFormState: React.Dispatch<CustomAttributes>;
 }
 
 export const EditTab: React.FC<EditTabProps> = props => {
   const styles = useStyles();
 
-  const { sidebarId, dispatchState } = props;
+  const { sidebarId, dispatchState, formState, setFormState } = props;
```
Comment (micahgodbolt) [inline packages/react-components/theme-designer/src/components/Sidebar/EditTab.tsx:51]: I don't think that the form state should move out of this tab. this was causing unnecessary re-renders before

Comment (micahgodbolt) [reply packages/react-components/theme-designer/src/components/Sidebar/EditTab.tsx:51]: hmmm...seems the deploy site is working ok. still curious why this state needs to be moved up

Comment (micahgodbolt) [reply packages/react-components/theme-designer/src/components/Sidebar/EditTab.tsx:51]: if you need some 'defaultState' coming from the parent, that's different

**Comment by micahgodbolt** [comment]
this allows form state to persist across tab switching. i agree that this is a better bit of UI rather than losing your custom theme just by switching tabs. Checked the deploy site and i'm not seeing any of the focus loss issues we had in the past.

**Pull request merged**